### PR TITLE
Update input params

### PR DESCRIPTION
**Pull request by OrellBuehler**

### CODE REVIEW

File: tcat-app/inputparams.py [hunk 1]
```diff
@@ -1,39 +1,57 @@
...
     ],
-    ('launch_vehicle_configuration', 'Launch Vehicle Configuration'): [
-        ['select', 'text', 'launcher', 'Launch vehicle name and variant', 'Soyuz_2.1a_Fregat',
-         ['Ariane_62', 'Ariane_64', 'Soyuz_2.1a_Fregat', 'Soyuz_2.1b_Fregat', 'Soyuz']],
+    ('launcher', 'Launcher'): [
+        ['checkbox', '', 'launcher_use_database', 'Use database', True, [True, False]],
+        ['select', 'text', 'launcher', 'Launch vehicle name and variant', 'Soyuz_2.1a_Fregat', ['Ariane_62', 'Ariane_64', 'Soyuz_2.1a_Fregat', 'Soyuz_2.1b_Fregat', 'Soyuz']],
         ['select', 'text', 'launch_site', 'Launch Site', 'Baikonur', ['Korou', 'Baikonur']],
         ['select', 'text', 'orbit_type', 'Orbit Type', 'LEO', ['LEO', 'SSO', 'LPEO', 'Polar', 'MTO', 'GTO', 'GTO+']],
         ['input', 'number', 'dispenser_tech_level', 'Dispenser technological level', 1.1, [0.1, 1.9]],
```
Comment (MatUspace) [inline tcat-app/inputparams.py]: dispenser tech level is not required anymore

File: tcat-app/inputparams.py [hunk 2]
```diff
@@ -1,39 +1,57 @@
 # array structure [control_name , datatype , lookup_name , speaking_name , default_value , accepted_values]
 
 constellation_mission_params = {
-    ('mission_configuration', 'Mission Configuration'): [
-        ['select', 'text', 'propulsion_type', 'Propulsion type', 'electrical',
-         ['electrical', 'water', 'mono-propellant', 'solid', 'bi-propellant']],
+    ('scenario_metadata', 'Scenario Metadata'): [
```
Comment (MatUspace) [inline tcat-app/inputparams.py:4]: keep "mission configuration" (and the word "configuration" in all title, sorry Malo did not use the correct wording in the email)

File: tcat-app/inputparams.py [hunk 3]
```diff
@@ -1,39 +1,57 @@
...
         ['checkbox', '', 'verbose', 'Debug (display figures)', False, [True, False]],
-        ['input', 'datetime-local', 'starting_epoch', 'Starting Epoch', '2025-01-01T12:00',
-         ['2000-01-01T00:00', '2100-01-01T00:00']],
-        ['input', 'number', 'mission_cash_limitor', 'Mission Cash Limitor', 0.01, [0.0, 1.0]],
+        ['input', 'datetime-local', 'starting_epoch', 'Starting Epoch', '2025-01-01T12:00', ['2000-01-01T00:00', '2100-01-01T00:00']],
+        ['input', 'number', 'tradeoff_mission_price_vs_duration', 'Tradeoff Mission Price vs Duration', 0.01, [0.0, 1.0]],
     ],
-    ('constellation_configuration', 'Constellation Configuration'): [
```
Comment (MatUspace) [inline tcat-app/inputparams.py:12]: keep "constellation configuration" (same in ADR scenario)

File: tcat-app/inputparams.py (hunk 1 above)
Comment (MatUspace) [reply tcat-app/inputparams.py]: strange it appears here in black but is not in the file when I open without the review, so apparently it's already removed

File: tcat-app/inputparams.py [hunk 4]
```diff
@@ -1,39 +1,57 @@
...
+        ['input', 'number', 'launcher_performance', 'Custom performance (kg)', None, [0.1, 99999.9]],
+        ['input', 'number', 'launcher_fairing_diameter', 'Fairing diameter (m)', None, [0.1, 100.0]],
+        ['input', 'number', 'launcher_fairing_cylinder_height', 'Fairing height of the cylindrical section (m)', None, [0.1, 100.0]],
+        ['input', 'number', 'launcher_fairing_total_height', 'Fairing total height (m)', None, [0.1, 100.0]],
+        ['select', 'text', 'launcher_perf_interpolation_method', 'Interpolation method for the launch vehicle performance', 'linear', ['linear', 'nearest', 'cubic', 'nearest-up', 'zero', 'slinear', 'quadratic', 'previous', 'next']]
     ],
-    ('orbits_configuration', 'Orbits Configuration'): [
+    ('kickstage', 'Kickstage'): [
+        ['checkbox', '', 'kickstage_use_database', 'Use database', True, [True, False]],
+        ['input', 'text', 'kickstage_name', 'Name', 'My Kickstage', '^[A-Za-z0-9 ]+$'],
```
Comment (MatUspace) [inline tcat-app/inputparams.py]: just like for launchers, need here a drop down list with the kick stages existing in the database : [Fregat, Fregat_M, AVUM_plus]

File: tcat-app/inputparams.py [hunk 5]
```diff
@@ -81,23 +96,44 @@
...
+        ['input', 'number', 'servicer_initial_fuel_mass', 'Fuel mass (kg)', 0.0, [0.0, 9999999]],
+        ['input', 'number', 'servicer_capture_module_dry_mass', 'Capture module dry mass (kg)', 0.0, [0.0, 9999999]],
+        ['input', 'number', 'servicer_prop_thrust', 'Prop thrust (N)', 0.0, [0.0, 9999999]],
+        ['input', 'number', 'servicer_prop_isp', 'Prop isp (s)', 0.0, [0.0, 9999999]],
+        ['input', 'number', 'servicer_propulsion_dry_mass', 'Propulsion dry mass (kg)', 0.0, [0.0, 9999999]],
+        ['input', 'number', 'servicer_struct_mass', 'Struct mass (kg)', 0.0, [0.0, 9999999]],
+        ['input', 'number', 'servicer_default_volume', 'Default volume (m3)', 0.0, [0.0, 9999999]],
+        ['select', 'text', 'servicer_propulsion_type', 'Propulsion type', 'mono-propellant', ['mono-propellant', 'water', 'solid', 'bi-propellant', 'electrical']],
     ],
-    ('orbits_configuration', 'Orbits Configuration'): [
```
Comment (MatUspace) [inline tcat-app/inputparams.py:100]: keep "orbit configuration" (Same for constellation deployment)

**Review by MatUspace — CHANGES_REQUESTED** [review]

File: tcat-app/static/js/inputparams_kickstage_conditions.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    kickstageDispenserDryMassLabel,
+    kickstageStructMassLabel,
+    kickstagePropulsionTypeLabel,
+]
+
+document.addEventListener("DOMContentLoaded", () => {
+    setRequiredKickstageFields();
+});
+
+if(kickstageUseDatabaseElement) {
```
Comment (MatUspace) [inline tcat-app/static/js/inputparams_kickstage_conditions.js:52]: I don't see where the differences of required inputs is set between using the DB or the custom fields

**Review by MatUspace — APPROVED** [review]

File: tcat-app/static/js/inputparams_kickstage_conditions.js (hunk 6 above)
Comment (OrellBuehler) [reply tcat-app/static/js/inputparams_kickstage_conditions.js:52]: I am not sure what mean?

Comment (MatUspace) [reply tcat-app/static/js/inputparams_kickstage_conditions.js:52]: is it here that we set which inputs are required depending on if the user takes a vehicle from the DB or if they provide custom data ?

Comment (OrellBuehler) [reply tcat-app/static/js/inputparams_kickstage_conditions.js:52]: yes

**Pull request merged**